jangjuseong/app.js

```diff
@@ -50,14 +50,148 @@ app.post('/users/signup', async (req, res) => {
         ?, 
         ?, 
         ?
-        );
+      );
     `,
     [name, email, password, profileImage]
   );
 
   res.status(201).json({ message: 'userCreated' });
 });
 
+app.post('/posts', async (req, res) => {
+  const { title, content, postImgUrl, userId } = req.body;
+
+  await mysqlDataSource.query(
+    `INSERT INTO posts (
+      title,
+      content,
+      post_image_url,
+      user_id
+      )
+      VALUES (
+        ?,
+        ?,
+        ?,
+        ?
+      );
+        `,
+    [title, content, postImgUrl, userId]
+  );
+
+  res.status(201).json({ message: 'postCreated' });
+});
+
+app.get('/posts', async (req, res) => {
+  const posts = await mysqlDataSource.query(
+    `SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.post_image_url AS postingImageUrl,
+      p.content AS postingContent
+    FROM posts p
+    INNER JOIN users u ON u.id = p.user_id;
+    `
+  );
+
+  res.status(200).json({ posts });
+});
+
+app.get('/posts/user/:userId', async (req, res) => {
+  const { userId } = req.params;
+
+  await mysqlDataSource
+    .query(
+      `SELECT
+    u.id AS userId,
+    u.profile_image AS userProfileImage,
+    pi.post_informations AS postings
+    FROM
+      users u
+    INNER JOIN (
+      SELECT
+        user_id,
+        JSON_ARRAYAGG(
+          JSON_OBJECT (
+            "postingId", id,
+            "postingImageUrl", post_image_url,
+            "postingContent", content
+          )
+        ) AS post_informations
+      FROM
+        posts
+      GROUP BY
+        user_id
+    ) pi 
+      ON pi.user_id = u.id
+    WHERE
+      u.id = ${userId};
+    `
+    )
+    .then((data) => res.status(200).json({ data }));
+});
+
+app.patch('/posts/:postId/user/:userId', async (req, res) => {
+  const { userId, postId } = req.params;
+  const { content } = req.body;
+  await mysqlDataSource.query(
+    `UPDATE posts
+    SET
+      content = ?
+    WHERE 
+      user_id = ${userId} AND id = ${postId}
+    `,
+    [content]
+  );
+
+  await mysqlDataSource
+    .query(
+      `SELECT
+      u.id AS userId,
+      u.profile_image AS userProfileImage,
+      p.id AS postingId,
+      p.post_image_url AS postingImageUrl,
+      p.content AS postingContent
+    FROM posts p
+    INNER JOIN users u ON u.id = p.user_id
+    WHERE u.id=${userId} AND p.id=${postId}
+    `
+    )
+    .then((data) => res.status(201).json({ data }));
```

Comment:
- 해당 부분도 await 키워드를 사용하면서 then() 함수로 비동기 함수를 처리하고 있네요!
- 포매팅까지 수정해주면 너무 좋을 것 같습니다.
- `?` 를 활용하여 데이터 매핑을 할 수 있도록 수정해주세요!

```suggestion
  const [result] = await mysqlDataSource.query(`
    SELECT
      u.id AS userId,
      u.profile_image AS userProfileImage,
      p.id AS postingId,
      p.post_image_url AS postingImageUrl,
      p.content AS postingContent
    FROM posts p
    INNER JOIN users u ON u.id = p.user_id
    WHERE u.id = ? AND p.id = ?
    `, [ userId, postId ])
   
   return res.status(201).json({ data: result }));
```